internal/translayer/dbuser/dbuser.go

```diff
@@ -0,0 +1,74 @@
+package dbuser
+
+import (
+	"context"
+	"errors"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+var (
+	// ErrorNotFound is returned when an database user was not found
+	ErrorNotFound = errors.New("database user not found")
+)
+
+type Service struct {
```

Comment:
I am still hesitant about the term "Service" here. The methods are concrete so my intuition is that the struct should also be concrete, i.e. `AtlasDatabaseUsers` that (maybe) implements some common "Service" interface.

`AtlasDatabaseUsers.Get`, `AtlasDatabaseUsers.Delete`, `AtlasDatabaseUsers.Exists`, etc. then have a concrete meaning.